### PR TITLE
Cleanup OC 

### PR DESCRIPTION
**Pull request by smisidjan**

### CODE REVIEW

File: Installation/Cronjob/oc.githubScrapper.cronjob.json [hunk 1]
```diff
@@ -3,10 +3,9 @@
     "$id": "https://opencatalogi.nl/cronjob/oc.githubScrapper.cronjob.json",
     "$schema": "https://docs.commongateway.nl/schemas/Cronjob.schema.json",
     "version": "0.0.1",
```
Comment (WilcoLouwerse) [inline Installation/Cronjob/oc.githubScrapper.cronjob.json]: You might want/need to upgrade this version to 0.0.2

File: Installation/Cronjob/oc.openCatalogi.cronjob.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "$id": "https://opencatalogi.nl/cronjob/oc.openCatalogi.cronjob.json",
     "$schema": "https://docs.commongateway.nl/schemas/Cronjob.schema.json",
     "version": "0.0.1",
```
Comment (WilcoLouwerse) [inline Installation/Cronjob/oc.openCatalogi.cronjob.json]: You might want/need to upgrade this version to 0.0.2

File: Installation/Cronjob/oc.bronnenTrigger.cronjob.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "$id": "https://opencatalogi.nl/cronjob/oc.bronnenTrigger.cronjob.json",
     "$schema": "https://docs.commongateway.nl/schemas/Cronjob.schema.json",
     "version": "0.0.1",
```
Comment (WilcoLouwerse) [inline Installation/Cronjob/oc.bronnenTrigger.cronjob.json]: You might want/need to upgrade this version to 0.0.2

File: src/ActionHandler/ComponentenCatalogusApplicationToGatewayHandler.php [hunk 4]
```diff
@@ -56,7 +105,7 @@ public function getConfiguration(): array
      */
     public function run(array $data, array $configuration): array
     {
-        return $this->service->getApplications();
+        return $this->service->getComponentenCatalogusApplications($data, $configuration);
```
Comment (WilcoLouwerse) [inline src/ActionHandler/ComponentenCatalogusApplicationToGatewayHandler.php]: That's a very long function name

File: src/ActionHandler/ComponentenCatalogusComponentToGatewayHandler.php [hunk 5]
```diff
@@ -56,7 +89,7 @@ public function getConfiguration(): array
      */
     public function run(array $data, array $configuration): array
     {
-        return $this->service->getComponents();
+        return $this->service->getComponentenCatalogusComponents($data, $configuration);
```
Comment (WilcoLouwerse) [inline src/ActionHandler/ComponentenCatalogusComponentToGatewayHandler.php]: That's a very long function name

File: src/ActionHandler/DeveloperOverheidApiToGatewayHandler.php [hunk 6]
```diff
@@ -56,7 +89,7 @@ public function getConfiguration(): array
      */
     public function run(array $data, array $configuration): array
     {
-        return $this->service->getComponents();
+        return $this->service->getDeveloperOverheidComponents($data, $configuration);
```
Comment (WilcoLouwerse) [inline src/ActionHandler/DeveloperOverheidApiToGatewayHandler.php]: That's a very long function name

File: src/ActionHandler/DeveloperOverheidRepositoryToGatewayHandler.php [hunk 7]
```diff
@@ -56,7 +81,7 @@ public function getConfiguration(): array
      */
     public function run(array $data, array $configuration): array
     {
-        return $this->service->getRepositories();
+        return $this->service->getDeveloperOverheidRepositories($data, $configuration);
```
Comment (WilcoLouwerse) [inline src/ActionHandler/DeveloperOverheidRepositoryToGatewayHandler.php]: That's a very long function name

File: src/Service/GithubApiService.php [hunk 8]
```diff
@@ -86,6 +88,61 @@ public function __construct(
...
+            $response = $this->callService->call($source, '/repos/'.$name);
+        } catch (ClientException $exception) {
+            $this->pluginLogger->error($exception->getMessage(), ['plugin' => 'open-catalogi/open-catalogi-bundle']);
+        }
+
+        if (isset($response) === false) {
+            return null;
+        }
+
+        $repository = json_decode($response->getBody()->getContents(), true);
```
Comment (rjzondervan) [inline src/Service/GithubApiService.php]: Isn't it better to use \Safe\json_decode()?

**Review by rjzondervan — APPROVED** [review]
Found one json_decode instead of \Safe\json_decode.

File: src/Service/ComponentenCatalogusService.php [hunk 9]
```diff
@@ -21,343 +21,102 @@ class ComponentenCatalogusService
...
     {
-        // Do we have a source?
-        $source = $this->resourceService->getSource('https://opencatalogi.nl/source/oc.componentencatalogus.source.json', 'open-catalogi/open-catalogi-bundle');
+        $this->data          = $data;
+        $this->configuration = $configuration;
 
-        $this->pluginLogger->info('Getting application '.$applicationId);
-        $response = $this->callService->call($source, '/products/'.$applicationId);
+        // Get the source from the configuration array.
+        $source   = $this->resourceService->getSource($this->configuration['source'], 'open-catalogi/open-catalogi-bundle');
```
Comment (WilcoLouwerse) [inline src/Service/ComponentenCatalogusService.php:83]: Just a reminder that the resourceService functions can return null and $source = null (or any other case like this ending up in a null result) can break your code.

File: src/Service/ImportResourcesService.php [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+    }//end importRepositoryThroughComponent()
+
+
+    /**
+     * @param array        $componentArray  The component array to import.
+     * @param ObjectEntity $componentObject The resulting component object.
+     *
+     * @return ObjectEntity|null
+     */
+    public function importLegalRepoOwnerThroughComponent(array $componentArray, ObjectEntity $componentObject): ?ObjectEntity
```
Comment (WilcoLouwerse) [inline src/Service/ImportResourcesService.php:147]: missing description in docblock

**Review by WilcoLouwerse — APPROVED** [review]
a few small comments

**Review by rjzondervan — APPROVED** [review]

**Pull request merged**